rspotify-model/src/idtypes.rs

```diff
@@ -149,99 +114,57 @@ pub enum IdError {
     InvalidId,
 }
 
-impl<T: IdType> std::fmt::Display for &Id<T> {
-    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
-        write!(f, "spotify:{}:{}", T::TYPE, &self.id)
-    }
-}
-
-impl<T> AsRef<str> for &Id<T> {
-    fn as_ref(&self) -> &str {
-        &self.id
-    }
-}
-
-impl<T> Borrow<str> for &Id<T> {
-    fn borrow(&self) -> &str {
-        &self.id
-    }
-}
-
-impl<T: IdType> std::str::FromStr for IdBuf<T> {
-    type Err = IdError;
-
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
-        Id::from_id_or_uri(s).map(|id| id.to_owned())
-    }
-}
-
-impl<T: IdType> Id<T> {
-    /// Owned version of the id [`IdBuf`](crate::idtypes::IdBuf).
-    pub fn to_owned(&self) -> IdBuf<T> {
-        IdBuf {
-            _type: PhantomData,
-            id: (&self.id).to_owned(),
-        }
-    }
-
-    /// Spotify object type
-    pub fn _type(&self) -> Type {
-        T::TYPE
-    }
-
-    /// Spotify object id (guaranteed to be a string of alphanumeric characters)
-    pub fn id(&self) -> &str {
-        &self.id
-    }
+/// The main interface for an ID.
+///
+/// # Implementation note
+///
+/// Note that for IDs to be useful their trait must be object-safe. Otherwise,
+/// it wouldn't be possible to use `Vec<Box<dyn Id>>` to take multiple kinds of
+/// IDs or just `Box<dyn Id>` in case the type wasn't known at compile time.
+/// This is why this trait includes both [`Self::_type`] and
+/// [`Self::_type_static`], and why all of the static methods use `where Self:
+/// Sized`.
```

Comment:
Then, after including the `Self::_type` and `Self::_type_static` for `Id`, the compiler can deduce the specific type of `Id`?

Reply:
These methods are just getters for the Id's type just like what `Id::_type` was before this PR. See the next comment as to why `_type_static` is necessary.